presto-teradata-functions/src/main/java/com/facebook/presto/teradata/functions/TeradataDateFunctions.java

```diff
@@ -78,19 +77,11 @@ public static Slice toChar(
     @Description("Converts a string to a DATE data type")
     @ScalarFunction("to_date")
     @SqlType(StandardTypes.DATE)
-    public static long toDate(
-            @OperatorDependency(
-                    operator = OperatorType.CAST,
-                    returnType = StandardTypes.DATE,
-                    argumentTypes = StandardTypes.TIMESTAMP)
-                    MethodHandle castToDate,
-            ConnectorSession session,
-            @SqlType(StandardTypes.VARCHAR) Slice dateTime,
-            @SqlType(StandardTypes.VARCHAR) Slice formatString)
+    public static long toDate(ConnectorSession session, @SqlType(StandardTypes.VARCHAR) Slice dateTime, @SqlType(StandardTypes.VARCHAR) Slice formatString)
     {
         try {
-            long millis = parseMillis(session, dateTime, formatString);
-            return (long) castToDate.invokeExact(session, millis);
+            long millis = parseMillis(UTC_KEY, session.getLocale(), dateTime, formatString);
```

Comment:
Shouldn't passing `UTC_KEY` depend on legacy timestamp flag?

Reply:
No new tests are needed?

Reply:
> Shouldn't passing UTC_KEY/session zone depend on legacy timestamp flag?

DATEs have just one semantics (no legacy here)

> No new tests are needed?

Tests exists and implementation was working correctly. This just simplifies.